python/ray/data/tests/test_batcher.py

```diff
@@ -127,6 +129,45 @@ def next_and_check(
     )
 
 
+# Test for 3 cases
+# 1. Batch size is less than block size
+# 2. Batch size is more than block size
+# 3. Block size is not divisble by batch size
+@pytest.mark.parametrize("batch_size", [4, 16, 7])
+def test_async_batcher(batch_size):
```

Comment:
1. test different queue size
2. test timeout and make sure it works
3. enable config by default in 1 commit to make sure no regression
4. enable for batch prediction by default and run nightly test

Reply:
very large block (10k rows), batch size of 10, run through map_batches and iter_batches. udf is time.sleep

make sure timing makes sense (== udf time * 1000)